02-express-tutorial/final/09-middleware-use.js

```diff
@@ -22,3 +22,37 @@ app.get('/api/items', (req, res) => {
 app.listen(5000, () => {
   console.log('Server is listening on port 5000....')
 })
+
+
+
+// const express = require('express')
+// const app = express()
+// const logger = require('./logger')
+// const authorize = require('./authorize')
+// // req => middlewar => res
+
+// app.use([logger, authorize])// use array to execute multiple middleware functions
+// //api/home/about/products
+
+// app.get('/',  (req,res) => {//path and callback function
+ 
+//     res.send('Home')
```

Comment:
It can get confusing between when to use use app.use(). This function is used to use middleware in express. Middleware can be custom built like logger , it can be built-in middleware that express has , or it can be third party ones such as morgan" etc . Mogan is a third party module that does similar work as logger function did in this example.